lib/index.config.js

```diff
@@ -23,36 +23,20 @@ export default function routes($ocLazyLoadProvider, $stateProvider, $translatePr
 		url: '/page1',
 		views: {
 			mainView: {
-				templateProvider: ($q) => {
-					return $q((resolve) => {
-						// lazy load the view
-						require.ensure([], () => resolve(require("./components/page1/page1.html")));
-					});
-				},
-				controller: function () {
-					this.name = 'Hello Page1';
-				}
+				component: 'page1'
 			},
 			rightMenuView: {
-				templateProvider: ($q) => {
-					return $q((resolve) => {
-						// lazy load the view
-						require.ensure([], () => resolve(require("./components/rightMenu/rightMenu.html")));
-					});
-				}
+				component: 'rightMenu'
 			}
 		},
+		resolvePolicy: {
+			module: {when: 'EAGER'}
```

Comment:
In version 1.0.0 final, view load process will happen later, so you can just use the onEnter hook